sdcm/cluster.py

```diff
@@ -4385,6 +4389,34 @@ def stop_nemesis(self, timeout=10):
                 stack_trace = traceback.format_stack(current_thread_frames[nemesis_thread.ident])
                 threads_tracebacks.append("\n".join(stack_trace))
 
+    def start_kms_key_rotation_thread(self, rotation_interval_m=60) -> None:
+        if self.params.get("cluster_backend") != 'aws':
+            return None
+        kms_key_alias_name = ""
+        append_scylla_yaml = yaml.safe_load(self.params.get("append_scylla_yaml")) or {}
+        for kms_host_name, kms_host_data in append_scylla_yaml.get("kms_hosts", {}).items():
+            if "auto" in kms_host_name:
+                kms_key_alias_name = kms_host_data["master_key"]
+                break
+        if not kms_key_alias_name:
+            self.log.debug("Automatic KMS key is not configured. Skip creation of it's rotation thread.")
+            return None
+
+        def _rotate_kms_key(kms_key_alias_name, rotation_interval_m):
+            while True:
+                time.sleep(rotation_interval_m * 60)
+                aws_kms.rotate_kms_key(kms_key_alias_name=kms_key_alias_name)
+
+        self.log.info("Start KMS key rotation thread for the '%s' alias", kms_key_alias_name)
+        aws_kms = AwsKms(region_names=self.params.region_names)
+        kms_key_rotation_thread = threading.Thread(
+            name='KmsKeyRotationThread',
+            target=_rotate_kms_key,
+            kwargs={"kms_key_alias_name": kms_key_alias_name, "rotation_interval_m": rotation_interval_m},
+            daemon=True)
+        kms_key_rotation_thread.start()
```

Comment:
shouldn't we implement something to stop this thread ? on teardown of the test ?

Reply:
I don't see practical need here. It sleeps for long time and then makes just 1 API call.

Reply:
if an api call would fail, the whole thread would stop, and we won't notice it.

Reply:
failures should go unnoticed

Reply:
I would argue the while loop should have a try/except clause

Reply:
> failures should go unnoticed

The `not`-word-thief has returned!

> I would argue the while loop should have a try/except clause

A can add try-block, not a problem.